mlflow/store/db_migrations/versions/2c7e13772396_convert_experiment_id_to_fixed_length_.py

```diff
@@ -0,0 +1,208 @@
+"""convert experiment id to fixed length random integer
+
+Revision ID: 2c7e13772396
+Revises: 97727af70f4d
+Create Date: 2022-10-18 16:39:31.183846
+
+This migration converts the primary key `experiment_id` within the Experiments table from an
+auto-incrementing primary key to a non-nullable unique-constrained Integer column. This is to
+support concurrent experiment creation and avoid collisions.
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy import PrimaryKeyConstraint, ForeignKeyConstraint, Sequence
+from sqlalchemy.inspection import inspect
+import logging
+
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.INFO)
+
+# revision identifiers, used by Alembic.
+revision = "2c7e13772396"
+down_revision = "97727af70f4d"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
```

Comment:
Given the complexity of this migration, do we have test coverage that ensures a migration from 1.30 -> 2.0 succeeds on all three databases with preexisting data in the `experiments` and `runs` table?

If such a test doesn't already exist, I'm admittedly a bit concerned that we might miss something in this migration and block folks from upgrading to MLflow 2.0. Given that there doesn't seem to be a big motivation for this change beyond creating a consistent experience with FileStore (SQLAlchemy store already has nice concurrency properties for experiment operations), I might suggest we hold off unless the test coverage is already present. Let me know what you think here.

Reply:
This is a very valid point. We could add a new test suite that validates migration backwards compatibility. I feel as though the time commitment to building that out is significantly higher than the effort involved in this PR. 
How about we hold off on any further work here until when/if it becomes an actual necessity (unless we think that pursuing a migration validation test suites is worthwhile for 2.0?).